### PR TITLE
MAP-1130 Format location description and use this for sorting

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/resources/db/migration/data/R__3_2__AGENCY_INTERNAL_LOCATIONS.sql [hunk 1]
```diff
@@ -36,8 +36,8 @@ INSERT INTO AGENCY_INTERNAL_LOCATIONS (INTERNAL_LOCATION_ID, INTERNAL_LOCATION_C
 
 
 INSERT INTO AGENCY_INTERNAL_LOCATIONS (INTERNAL_LOCATION_ID, INTERNAL_LOCATION_CODE, INTERNAL_LOCATION_TYPE, AGY_LOC_ID, UNIT_TYPE, CERTIFIED_FLAG, CAPACITY, OPERATION_CAPACITY, DESCRIPTION, PARENT_INTERNAL_LOCATION_ID, USER_DESC, NO_OF_OCCUPANT)
- VALUES (-1,   'A',   'WING',  'LEI',   'NA', 'Y', 14, 13, 'LEI-A', null, 'Block A', 8),
-        (-2,   '1',   'LAND',  'LEI',   'NA', 'Y', 14, 13, 'LEI-A-1', -1, 'Landing A/1', 7),
+ VALUES (-1,   'A',   'WING',  'LEI',   'NA', 'Y', 14, 13, 'LEI-A', null, 'BLOCK A', 8),
```
Comment (Mjwillis) [inline src/main/resources/db/migration/data/R__3_2__AGENCY_INTERNAL_LOCATIONS.sql:39]: You can't have lowercase in NOMIS but we want to show in sentence case

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/MovementsRepositorySql.kt [hunk 2]
```diff
@@ -209,7 +209,7 @@ enum class MovementsRepositorySql(val sql: String) {
         AIL.INTERNAL_LOCATION_CODE                            AS LOCATION_CODE,
         AIL.DESCRIPTION                                       AS FULL_LOCATION_PATH,
         (CASE 
-	        WHEN AIL.INTERNAL_LOCATION_TYPE IN ('CELL', 'ROOM') THEN AIL.INTERNAL_LOCATION_CODE 
+	        WHEN AIL.INTERNAL_LOCATION_TYPE = 'CELL' THEN AIL.INTERNAL_LOCATION_CODE
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/sql/MovementsRepositorySql.kt:212]: Don't want to do this for a room

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/RollCountDto.kt [hunk 3]
```diff
@@ -28,7 +28,7 @@ data class RollCountDto(
     this.locationType,
     this.locationCode,
     LocationProcessor.stripAgencyId(this.fullLocationPath, prisonId),
-    this.livingUnitDesc,
+    LocationProcessor.formatLocation(this.livingUnitDesc),
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/api/model/RollCountDto.kt:31]: Format the location

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/RollCount.java [hunk 4]
```diff
@@ -103,6 +103,6 @@ public class RollCount implements SortAttribute {
     @org.jetbrains.annotations.NotNull
     @Override
     public String getKey() {
-        return getFullLocationPath();
+        return getLivingUnitDesc();
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/api/model/RollCount.java:106]: Sort now by the description (or code)

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**